tasks/test.rake

```diff
@@ -42,9 +42,51 @@ task 'test:spec' do
 end
 
 ##
-## feature / integration tests
+## integration / smoke tests
 ##
 
+desc 'Executes smoke tests for a single gem'
+rule /^test:smoke:.+$/ do |task|
+  dir = "gems/#{task.name.split(':').last}/features"
+  tags = "-t '@smoke'"
+  ENV.fetch('AWS_SMOKE_TEST_SKIP_TAGS', '').split(',').each do |tag|
+    tags += " -t 'not @#{tag}'"
+  end
+  sh("bundle exec cucumber --retry 3 #{tags} -r #{dir} #{dir} --publish-quiet")
+end
+
+desc 'Executes all smoke tests'
+rule 'test:smoke' do
+  failures = []
+  Dir.glob('gems/*/features').each do |dir|
```

Comment:
I think we should skip the gem if the features directory doesn't contain any *.feature files.  It ends up adding a lot of time to the total to spin up a new process for every service just to run 0 tests for it.

Reply:
Yeah, that's a good idea, done